packages/core/src/electron-browser/menu/electron-main-menu-factory.ts

```diff
@@ -11,13 +11,14 @@ import {
     CommandRegistry, isOSX, ActionMenuNode, CompositeMenuNode,
     MAIN_MENU_BAR, MenuModelRegistry, MenuPath
 } from '../../common';
-import { KeybindingRegistry, Keybinding, KeyCode, Key } from '../../browser';
+import { PreferenceService, KeybindingRegistry, Keybinding, KeyCode, Key } from '../../browser';
 
 @injectable()
 export class ElectronMainMenuFactory {
 
     constructor(
         @inject(CommandRegistry) protected readonly commandRegistry: CommandRegistry,
+        @inject(PreferenceService) protected readonly preferencesService: PreferenceService,
```

Comment:
There should be a solution without coupling with `PreferenceService`. How about keeping Electron support of the scope for now till we have a simple generic solution?

Reply:
If I remove this block, autosave menu item will receive not actual answers from `isToggled` request because this request reads preferences from cash, so `isToggled` call directly after changing preferences file will return not actual state. This blocks updates the menu item when it receives actual state.

Reply:
I meant that it feels like a workaround because other command handlers are won't necessary rely on preferences. I wonder there is a way to render the whole menu before it is shown.

Reply:
Agree, but I haven't found a way how to trigger show electon's menu event